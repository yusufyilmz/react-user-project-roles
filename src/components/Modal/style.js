import styled from 'styled-components';

export const Modal = styled.div`
    position: fixed; 
    z-index: 1; 
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto; 
    background-color: rgb(0,0,0); 
    background-color: rgba(0,0,0,0.4); 
`;

Modal.displayName = 'Modal'

export const ModalContent = styled.div`
    margin: 40% auto;
    text-align: center;
    font-size: x-large;
    width: 60%; 
    padding: 15px;
    margin-bottom: 20px;
    border: 1px solid transparent;
    border-radius: 4px;
    border-color: #ebccd1;
    background-color: ${props => props.result === 'OK' ? '#28a745' :  '#f2dede'}
    color: ${props => props.result === 'OK' ? 'white' :  '#a94442'}
`
ModalContent.displayName = 'ModalContent'


export const ModalClose = styled.span`
    color: #aaa;
    float: right;
    font-size: 28px;
    font-weight: bold;
    cursor: pointer;
`

ModalClose.displayName = 'ModalClose'