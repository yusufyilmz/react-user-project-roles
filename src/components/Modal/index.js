import React from 'react';
import { Modal, ModalContent, ModalClose } from './style';

export default ({ message, closeModal, result }) => {
    return (
        <Modal >
            <ModalContent result={result}>
                <ModalClose onClick={closeModal}>&times;</ModalClose>
                <strong>{message}</strong>
            </ModalContent>
        </Modal>
    )
}




