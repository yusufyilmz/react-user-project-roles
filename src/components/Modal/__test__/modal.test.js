

import React from 'react';
import Modal from '../index';
import { shallow } from 'enzyme'


describe('MODAL COMPONENT', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            value : 'test',
            closeModal : jest.fn(),
            result : "SUCCESS",
            message : "test",
            type : 'text'
        }
      
        wrapper = shallow(
            <Modal  {...props}  />
        );
    });

    test('renders onclick correctly', () => {
        wrapper.find('ModalClose').simulate('click');
        expect(props.closeModal.mock.calls.length).toEqual(1);
    });

    test('snapchat correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
})
