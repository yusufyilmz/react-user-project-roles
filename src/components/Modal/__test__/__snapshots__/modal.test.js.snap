// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snapchat correctly 1`] = `
<styled.div>
  <styled.div
    result="SUCCESS"
  >
    <ModalClose
      onClick={[MockFunction]}
    >
      Ã—
    </ModalClose>
    <strong>
      test
    </strong>
  </styled.div>
</styled.div>
`;
